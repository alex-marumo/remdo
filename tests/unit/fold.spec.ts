import { getMinimizedState, loadEditorState } from "./common";
import { it } from "vitest";

it("fold", async ({ editor, expect, lexicalUpdate }) => {
  const { note0 } = loadEditorState(editor, "basic");
  expect(getMinimizedState(editor)).toMatchSnapshot("base");
  lexicalUpdate(() => (note0.fold = true));
  expect(getMinimizedState(editor)).toMatchSnapshot("folded");

  lexicalUpdate(() => (note0.fold = true)); //noop
  expect(getMinimizedState(editor)).toMatchSnapshot("folded");

  lexicalUpdate(() => (note0.fold = false));
  expect(getMinimizedState(editor)).toMatchSnapshot("base");
});

it("load folded", async ({ editor, expect }) => {
  const { note0 } = loadEditorState(editor, "folded");
  expect(getMinimizedState(editor)).toMatchSnapshot();
});

it("modify folded", async ({ editor, expect, lexicalUpdate }) => {
  const { note0 } = loadEditorState(editor, "folded");
  lexicalUpdate(() =>
    note0.lexicalNode.getFirstChild().setTextContent("note0 - modified")
  );
  expect(getMinimizedState(editor)).toMatchSnapshot();
});
