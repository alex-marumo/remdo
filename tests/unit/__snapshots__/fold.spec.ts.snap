// Vitest Snapshot v1

exports[`fold > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;

exports[`fold > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;

exports[`fold > folded 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - fold: true
      type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;

exports[`fold > folded 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - fold: true
      type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;

exports[`load folded 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - fold: true
      type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;

exports[`modify folded 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - fold: true
      type: listitem
      value: 1
      children:
      - text: note0 - modified
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: note1
            type: text
"
`;
